src/index.ts

```diff
@@ -53,6 +53,8 @@ function encodeLength(len: number, offset: number): Buffer {
  * @param stream - Is the input a stream (false by default)
  * @returns - returns decode Array of Buffers containg the original message
  **/
+export function decode(input: Buffer, stream?: boolean): Buffer;
+export function decode(input: Buffer[], stream?: boolean): Buffer[];

```

Comment:
What does this do respectively what is the reasoning for this (just for my understanding)?

Reply:
this is function overloading. So basically instead of using union type for input and output we can say to if its buffer in it's buffer out.